docs/docs/create-source-plugin.md

```diff
@@ -100,3 +100,56 @@ package provides a set of helper functions to generate Node objects with the
 required fields. This includes automatically generating fields like node IDs
 and the `contentDigest` MD5 hash, keeping your code focused on data gathering,
 not boilerplate.
+
+## Advanced
+
+### Adding relationships between nodes
+
+Gatsby source plugins not only create nodes, they also create relationships between nodes that are exposed to GraphQL queries.
+
+There are two ways of adding node relationships in Gatsby: (1) hierarchical (parent-child) or (2) foreign-key based. 
+
+#### Hierarchical relationships
+
+An example of a hierarchical relationship would be a `transformer` plugin changing a file node's markdown string into a `childMarkdownRemark` node that is attached to that fileNode. The markdown node doesn't make sense outside of the file it belongs to. 
```

Comment:
Perhaps it'd be clearer to give a specific example e.g. "the gatsby-transformer-remark plugin _transforming_ a parent File node's markdown string into a `MarkdownRemark` node.  The transformer plugin adds its newly created child node as a child to the parent node using the action `createParentChildLink`. The markdown node is completely derived from the parent File node so wouldn't ever exist if the parent File node hadn't been created."

Reply:
great!